src/swaps/liquality/LiqualitySwapProvider.ts

```diff
@@ -127,13 +128,22 @@ export class LiqualitySwapProvider extends SwapProvider {
     };
   }
 
-  public async newSwap({ network, walletId, quote: _quote }: SwapRequest<LiqualitySwapHistoryItem>) {
+  public async newSwap(swapRequest: SwapRequest<LiqualitySwapHistoryItem>) {
+    const approveTx = await this.approve(swapRequest, true);
+    const updates = approveTx !== null ? approveTx : await this.initiateSwap(swapRequest);
```

Comment:
I think this can be written: `const updates = approveTx ? approveTx : await this.initiateSwap(swapRequest);`

Reply:
or even better

` const updates = approveTx || (await this.initiateSwap(swapRequest));`